### PR TITLE
Code Monitors: add code to execute slack webhooks

### PR DESCRIPTION
**Pull request by camdencheek**
This adds the actual code to execute webhooks. Its footprint is a little larger than ideal because I also had to move the email code into the `backend` package to avoid circular dependencies.

There is still no way to create new webhooks, so this code is not executed outside of tests. 

Note that this isn't the copy-edited form of the slack message, so the actual message will be reviewed for content/visual design. This is mostly just a functional placeholder.

### CODE REVIEW

File: enterprise/internal/codemonitors/background/slack.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		return slack.NewSectionBlock(slack.NewTextBlockObject("mrkdwn", s, false, false), nil, nil)
+	}
+
+	// To see what this looks like:
+	// https://app.slack.com/block-kit-builder/T02FSM7DL#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22text%22:%22*New%20results%20for%20code%20monitor*%22,%22type%22:%22mrkdwn%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%225%20new%20results%20for%20query%20%60%60%60type:diff%20repo:github.com/sourcegraph/sourcegraph$%20BEGIN%20PRIVATE%20KEY%60%60%60%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22%3Chttps://sourcegraph.com/search?q=context:global+type:diff+repo:github.com/sourcegraph/sourcegraph%2524+BEGIN+PRIVATE+KEY&patternType=literal%7CView%20Search%20Results%3E%20%7C%20%3Chttps://sourcegraph.com/code-monitoring?visible=1%7CView%20Code%20Monitor%3E%22%7D%7D%5D%7D
+	return &slack.WebhookMessage{
+		Blocks: &slack.Blocks{BlockSet: []slack.Block{
+			newMarkdownSection(fmt.Sprintf("*New results for Code Monitor \"%s\"*", args.MonitorDescription)),
+			newMarkdownSection(fmt.Sprintf("%d new results for query: ```%s```", args.NumResults, args.Query)),
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply enterprise/internal/codemonitors/background/slack.go:31]: The query is currently retrievable from the URL, so I wasn't considering that secret information. I don't _think_ that's something we need to hide, but I'm also open to other opinions.

[the remark this answers is not in the document]
Comment (limitedmage) [reply enterprise/internal/codemonitors/background/slack.go:31]: Oh that is true, it's already in the emails, just hidden in the URL. I think this is OK, unless security team thinks othewise.

[the remark this answers is not in the document]
Comment (camdencheek) [reply enterprise/internal/codemonitors/background/slack.go:31]: cc @sourcegraph/security

File: enterprise/internal/codemonitors/background/slack.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/sourcegraph/sourcegraph/internal/httpcli"
+)
+
+func sendSlackNotification(ctx context.Context, url string, args actionArgs) error {
+	return postSlackWebhook(ctx, httpcli.ExternalDoer, url, slackPayload(args))
+}
+
+func slackPayload(args actionArgs) *slack.WebhookMessage {
+	newMarkdownSection := func(s string) slack.Block {
+		return slack.NewSectionBlock(slack.NewTextBlockObject("mrkdwn", s, false, false), nil, nil)
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply enterprise/internal/codemonitors/background/slack.go:23]: The significance is that `mrkdwn` is the slack-sanctioned keyword for a markdown-formatted block. Slck dsn't lke vwls.

File: enterprise/internal/codemonitors/background/slack.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andreeleuterio) [reply enterprise/internal/codemonitors/background/slack.go:31]: Late to the party.. it's all good with us considering that it doesn't have the same issue we saw in saved searches recently.

File: enterprise/internal/codemonitors/background/slack.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rvantonder) [reply enterprise/internal/codemonitors/background/slack.go:23]: ll

**Pull request merged**